tests/cluster/tests/24-links.tcl

```diff
@@ -0,0 +1,106 @@
+        set num_links [llength $links]
+
+        # Two links per peer
+        assert {$num_peers*2 eq $num_links}
+
+        # For each peer there should be exactly one
+        # link "to" it and one link "from" it.
+        foreach n $nodes {
+            if {[has_flag $n myself]} continue
+            set peer [dict get $n id]
+            set to 0
+            set from 0
+            foreach l $links {
+                if {[dict get $l node] eq $peer} {
+                    if {[dict get $l dir] eq "to"} {
+                        incr to
+                    } elseif {[dict get $l dir] eq "from"} {
+                        incr from
+                    }
+                }
+            }
+            assert {$to eq 1}
+            assert {$from eq 1}
+        }
+    }
+}
+
+set primary1_id 0
+set primary2_id 1
+
+set primary1 [Rn $primary1_id]
+set primary2 [Rn $primary2_id]
+
+test "Disconnect link when send buffer limit reached" {
+    # Set timeout to 1 hour so links won't get disconnected due to timeouts
+    set_cluster_node_timeout [expr 60*60*1000]
+
+    # Get primary1's links with primary2
+    set primary2_name [dict get [get_myself $primary2_id] id]
+    set orig_link_p1_to_p2 [get_link_to_peer $primary1_id $primary2_name]
+    set orig_link_p1_from_p2 [get_link_from_peer $primary1_id $primary2_name]
+
+    # On primary1, set cluster link send buffer limit to 32MB
+    catch {$primary1 CONFIG set cluster-link-sendbuf-limit [expr 32*1024*1024]}
+
+    # To manufacture an ever-growing send buffer from primary1 to primary2,
+    # make primary2 unresponsive by running an infinite loop Lua script on it.
+    # On primary2, set Lua script time out to 1 hour
+    catch {$primary2 CONFIG set lua-time-limit 3600000}
+    set primary2_port [RPort $primary2_id]
+    set primary2_pid [get_info_field [$primary2 info] process_id]
+    exec ../../../src/redis-cli -p $primary2_port eval "while true do end" 0 &
+
+    # On primary1, send a 10MB Pubsub message. It will stay in send buffer of
+    # the link from primary1 to primary2
+    catch {$primary1 publish channel [prepare_value [expr 10*1024*1024]]}
+
+    # Check the same link has not been disconnected, but its send buffer has grown
+    set same_link_p1_to_p2 [get_link_to_peer $primary1_id $primary2_name]
+    assert {[dict get $same_link_p1_to_p2 ctime] eq [dict get $orig_link_p1_to_p2 ctime]}
+    assert {[dict get $same_link_p1_to_p2 sbufalloc] > [dict get $orig_link_p1_to_p2 sbufalloc]}
+
+    # On primary1, send another 10MB Pubsub message
+    catch {$primary1 publish channel [prepare_value [expr 10*1024*1024]]}
+
+    # Check the same link has not been disconnected, but its send buffer has grown again
+    set still_same_link_p1_to_p2 [get_link_to_peer $primary1_id $primary2_name]
+    assert {[dict get $still_same_link_p1_to_p2 ctime] eq [dict get $same_link_p1_to_p2 ctime]}
+    assert {[dict get $still_same_link_p1_to_p2 sbufalloc] > [dict get $same_link_p1_to_p2 sbufalloc]}
```

Comment:
Because each 10MB growth is due to a single command, I was paranoid that if I only test one 10MB, it might pass accidentally due to timing. E.g. 

1. Test sends `publish`
2. Test asserts p1->p2 link buffer has grown successfully.
3. `publish` is actually sent from p1 to p2 (p2 is not `SIGSTOP`-ed somehow).

I thought if I tested twice, this is less likely to happen.

I don't feel strongly so I removed the 2nd 10MB.